clustered/client/src/main/java/org/ehcache/clustered/client/internal/loaderwriter/writebehind/ClusteredWriteBehindStore.java

```diff
@@ -0,0 +1,385 @@
+      ByteBuffer payload = codec.encode(operation);
+      long extractedKey = extractLongKey(key);
+      Chain chain = storeProxy.getAndAppend(extractedKey, payload);
+      ResolvedChain<K, V> resolvedChain = resolver.resolve(chain, key, timeSource.getTimeMillis());
+
+      Result<K, V> result = resolvedChain.getResolvedResult(key);
+      return result == null ? null : result.getValue();
+    } catch (Exception re) {
+      throw handleException(re);
+    }
+  }
+
+  protected V silentReplace(K key, V oldValue, V newValue) throws StoreAccessException {
+    try {
+      ConditionalReplaceOperation<K, V> operation = new ConditionalReplaceOperation<>(key, oldValue, newValue, timeSource
+        .getTimeMillis());
+      ByteBuffer payload = codec.encode(operation);
+      long extractedKey = extractLongKey(key);
+      Chain chain = storeProxy.getAndAppend(extractedKey, payload);
+      ResolvedChain<K, V> resolvedChain = resolver.resolve(chain, key, timeSource.getTimeMillis());
+
+      Result<K, V> result = resolvedChain.getResolvedResult(key);
+      return result == null ? null : result.getValue();
+    } catch (Exception re) {
+      throw handleException(re);
+    }
+  }
+
+  public class WriteBehindServerCallback implements ServerStoreProxy.ServerCallback {
+
+    private final ServerStoreProxy.ServerCallback delegate;
+
+    WriteBehindServerCallback(ServerStoreProxy.ServerCallback delegate) {
+      this.delegate = delegate;
+    }
+
+    @Override
+    public void onInvalidateHash(long hash) {
+      this.delegate.onInvalidateHash(hash);
+    }
+
+    @Override
+    public void onInvalidateAll() {
+      this.delegate.onInvalidateAll();
+    }
+
+    @Override
+    public Chain compact(Chain chain) {
+      return this.delegate.compact(chain);
+    }
+
+    @Override
+    public Chain compact(Chain chain, long hash) {
+      clusteredWriteBehind.flushWriteBehindQueue(chain, hash);
+      return null;
+    }
+  }
+
+  private ServerStoreProxy.ServerCallback getWriteBehindServerCallback(ServerStoreProxy.ServerCallback delegate) {
+    return new WriteBehindServerCallback(delegate);
+  }
+
+  public class ClusteredWriteBehind {
+    private final ExecutorService executorService;
+    private final CacheLoaderWriter<? super K, V> cacheLoaderWriter;
+    private final OperationsCodec<K, V> codec;
+    private final ChainResolver<K, V> resolver;
+    private final TimeSource timeSource;
+
+    ClusteredWriteBehind(ExecutorService executorService,
+                         TimeSource timeSource, ChainResolver<K, V> resolver,
+                         CacheLoaderWriter<? super K, V> cacheLoaderWriter,
+                         OperationsCodec<K, V> codec) {
+      this.executorService = executorService;
+      this.resolver = resolver;
+      this.cacheLoaderWriter = cacheLoaderWriter;
+      this.codec = codec;
+      this.timeSource = timeSource;
+    }
+
+    void flushWriteBehindQueue(Chain ignored, long hash) {
```

Comment:
This method could be enhanced to respect batching configuration(like coalescing)